src/main/java/com/example/demo/ShortenUrlController.java

```diff
@@ -1,35 +1,65 @@
 package com.example.demo;
 
-import com.example.demo.dao.UrlRepository;
 import com.example.demo.model.Url;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.Optional;
 
 @RestController
+@RequestMapping("api/v1/")
 public class ShortenUrlController {
 
     @Autowired
-    private UrlRepository repository;
+    private final ShortenUrlService shortenUrlService;
 
-    @GetMapping
-    public String helloWorld(){
-        return "Hello World from Spring Boot";
+    @Autowired
+    public ShortenUrlController(final ShortenUrlService shortenUrlService){
+        this.shortenUrlService = shortenUrlService;
     }
 
+    //TODO: ShortenUrl Logic
     @PostMapping("/url")
-    public Url saveUrl(@RequestBody Url url){
+    public ResponseEntity<Void> saveUrl(@RequestBody Url url){
+        shortenUrlService.save(url);
```

Comment:
As we discussed offline there should be next logic

You are getting String longUrl

You are making shortUrl from Long

You are creating new object

Url url = new Url();
and setting all this fields to your collection